src/app/frontend/service/detail/controller.js

```diff
@@ -37,4 +41,35 @@ export class ServiceDetailController {
     /** @export {!angular.Resource} */
     this.endpointListResource = kdServiceEndpointResource;
   }
+
+  /**
+   * This func looks for user defined links that use the k8's api server proxy addresses. Since the
+   * default way of deploying/running dashboard is in a container we concluded that, for the time
+   * being, we will parse the host:port address from the current browser window used to access the
+   * k8's api server and inject it into the userlink proxy address so that its accessible from the
+   * browser running dashboard.
+   *
+   * @param {!backendApi.ServiceDetail} serviceDetail
+   * @param {!angular.$location} $location
+   * @return {!backendApi.ServiceDetail} serviceDetail
+   */
+  checkUserlinksForProxyURL(serviceDetail, $location) {
```

Comment:
This function will always create links - however the links are only correct if Dashboard is accessed through proxy. A maybe better alternative might be not to render the user links at all in case Dashboard is not accessed through proxy
However, as we consider this featrue alpha anyway it might be ok for now